### ISSUE TITLE
[Ray Client] Cannot connect to Ray Client when cluster doesn't have resource

### ISSUE THREAD

**Issue by simon-mo** [ray-project/ray#18136]
It seems like when the cluster runs out of CPUs, ray client connect hangs. And also submitting `num_cpus=0` task doesn't work. 

```
➜  ~ ray start --head --num-cpus 0
Local node IP: 192.168.1.69
2021-08-26 17:41:30,996	INFO services.py:1265 -- View the Ray dashboard at http://127.0.0.1:8265

--------------------
Ray runtime started.
--------------------

Next steps
  To connect to this Ray runtime from another node, run
    ray start --address='192.168.1.69:6379' --redis-password='5241590000000000'

  Alternatively, use the following Python code:
    import ray
    ray.init(address='auto', _redis_password='5241590000000000')

  To connect to this Ray runtime from outside of the cluster, for example to
  connect to a remote cluster from your laptop directly, use the following
  Python code:
    import ray
    ray.init(address='ray://<head_node_ip_address>:10001')

  If connection fails, check your firewall settings and network configuration.

  To terminate the Ray runtime, run
    ray stop
➜  ~ python
Python 3.7.11 (default, Jul 27 2021, 07:03:16)
[Clang 10.0.0 ] :: Anaconda, Inc. on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import ray
>>> ray.init(address="ray://localhost:10001")
The actor or task with ID 66b6c7ccbd8bfecc3c01ff786f64283ee81fbefd0eddfa7f cannot be scheduled right now. It requires {CPU: 1.000000} for placement, however the cluster currently cannot provide the requested resources. The required resources may be added as autoscaling takes place or placement groups are scheduled. Otherwise, consider reducing the resource requirements of the task.
```

1.6.0

**Comment by DmitriGekhtman** [comment]
Is this still an issue? If so, we should probably try to fix it. cc @ckw017

**Comment by ckw017** [comment]
This was fixed by #19604
